### PR TITLE
Fix qwidget errors

### PR DESCRIPTION
**Pull request by Hpn4**
avoid having a crash : Qwidget cannot be create before QApplication

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (Baschtien) [inline .pre-commit-config.yaml:1]: Is this wanted ?

**Review by Baschtien — APPROVED** [review]

File: .pre-commit-config.yaml
Comment (Hpn4) [reply .pre-commit-config.yaml:1]: yes ask simon

File: Holovibes/sources/gui/other/popup_error.cc [hunk 1]
```diff
@@ -5,15 +5,30 @@ namespace holovibes::gui
...
+        exit(exit_value);
+    }
+    else
+    {
+        LOG_ERROR(error_msg);
+        exit(exit_value);
+    }
 }
 
 void show_warn(const std::string& warn_msg)
```
Comment (Hpn4) [inline Holovibes/sources/gui/other/popup_error.cc:21]: Btw this methods is never called, delete ?

File: Holovibes/sources/gui/other/popup_error.cc [hunk 2]
```diff
@@ -5,15 +5,30 @@ namespace holovibes::gui
 {
 void show_error_and_exit(const std::string& error_msg, const int exit_value)
 {
-    QMessageBox messageBox;
-    messageBox.critical(nullptr, "Internal Error", error_msg.c_str());
-    exit(exit_value);
+    if (!holovibes::Holovibes::instance().is_cli)
```
Comment (Hpn4) [inline Holovibes/sources/gui/other/popup_error.cc:8]: Maybe add another condition to avoir the message to appear infinitly until the UI crashes. Or put the QMessage in static in order to have a single instance of it

File: Holovibes/sources/gui/other/popup_error.cc (hunk 1 above)
Comment (Hpn4) [reply Holovibes/sources/gui/other/popup_error.cc:21]: nop because arthur used it before, and it can be usefull so not for me

Comment (Hpn4) [reply Holovibes/sources/gui/other/popup_error.cc:21]: (from alexis)

File: Holovibes/sources/gui/other/popup_error.cc (hunk 2 above)
Comment (Hpn4) [reply Holovibes/sources/gui/other/popup_error.cc:8]: I don't understand what you are saying : it makes the program stop so there is no infinitly loop (alexis)

**Pull request merged**